src/marshmallow/fields.py

```diff
@@ -537,7 +537,21 @@ def _deserialize(self, value, attr, data, partial=None, **kwargs):
         return self._load(value, data, partial=partial)
 
 
-class List(Field):
+class ContainerMixin(object):
+    """Common methods for container fields"""
+
+    def get_container_modifiers(self, container):
```

Comment:
Drive-by: this name is a bit misleading, since this method actually mutates `self`.

Reply:
Yeah, I'm not really satisfied with those method names.

Reply:
`ContainerMixin.get_container_modifiers()` -> `ContainerMixin.set_modifiers()`?

`ContainerMixin.set_container_modifiers()` -> `Nested.set_modifiers()`?

It seems like `Nested` should be responsible for setting its own modifiers if other fields are. Making it a no-op on `Field` and just defining it where needed would avoid the type check and allow propagation between containers (like `List(List(Nested(S)))`).

Reply:
Seems like a reasonable approach.

Reply:
I pushed a new commit in which I tried to implement something along those lines. I still find it a bit awkward. But maybe I misunderstood your comments.

Reply:
I think Jared is suggesting to only have `set_modifiers` which container fields override.

```python
class Field(...):
    def set_modifiers(self):
        # noop

class ContainerMixin(...):

    def set_modifiers(self):
        # sets modifiers on inner fields
```

I think you can remove `get_modifiers`.